### PR TITLE
Connect react-redux , implement add, remove button

### PR DESCRIPTION
**Pull request by alanluqman**
## Description 📃 
In this Pull Request, I have connected React and Redux, and I have implemented Add and remove buttons.
for this purpose I have used these hooks ( `useState `, `useSelector `, `useDispatch `) and `Provider `as well.

### CODE REVIEW

**Review by ShahierNashaat — CHANGES_REQUESTED** [review]
Hi @alanluqman, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## To Highlight 💯 

- Correct use of GitFlow ✔️
- Descriptive commit messages ✔️
- Professional looking pull request ✔️
- Your readme looks professional ✔️
- Good overall design ✔️
- Your code is clean and well documented ✔️

## Required Changes ♻️
- [x] Kindly notice that the button for checking the status in the category component should dispatch the `underConstruction` action creator.

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by alanluqman** [comment]
Thank you so much @ShahierNashaat, I Have done with your request. 👍🏻

File: src/components/catigories.js [hunk 1]
```diff
@@ -1,7 +1,11 @@
+import { useDispatch } from 'react-redux';
+import { underConstruction } from '../redux/categories/categories';
+
 function Categories() {
```
Comment (Ismailco) [inline src/components/catigories.js]: - [x] [OPTIONAL] Your file name has a typo (catigories) kindly consider correcting it

File: src/components/catigories.js [hunk 2]
```diff
@@ -1,7 +1,11 @@
...
+  const dispatch = useDispatch();
   return (
     <>
-      <button type="button">Check Status</button>
+      <button type="button" onClick={() => dispatch(underConstruction)}>Check Status</button>
```
Comment (Ismailco) [inline src/components/catigories.js]: - [x] [OPTIONAL] Good job using dispatch on button click how ever you should display `Under construction` in the page not just in the store

**Review by Ismailco — APPROVED** [review]
### Hi @alanluqman, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

cr-approve


## Highlights

✔️  No linter errors
✔️ Used Gitflow
✔️ Work documented professionally

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: src/components/catigories.js (hunk 1 above)
Comment (alanluqman) [reply src/components/catigories.js]: Done 👍🏻

File: src/components/catigories.js (hunk 2 above)
Comment (alanluqman) [reply src/components/catigories.js]: Done 👍🏻

**Comment by alanluqman** [comment]
Thanks for your effort and suggestions @Ismailco , I did all 👍🏻

**Pull request merged**